### ISSUE TITLE
Parser erroring out on static method call

### ISSUE THREAD

**Issue by Lapz** [rust-analyzer/rust-analyzer#1187]
Given the following macro 
```rust
macro_rules! vec {
            ($($item:expr),*) => {
                {
                    let mut v = Vec::new();
                    $(
                        v.push($item);
                    )*
                    v
                }
            };
}
```

expansion will produce 
```
BLOCK_EXPR@[0; 45)
  BLOCK@[0; 45)
    L_CURLY@[0; 1) "{"
    LET_STMT@[1; 12)
      LET_KW@[1; 4) "let"
      BIND_PAT@[4; 8)
        MUT_KW@[4; 7) "mut"
        NAME@[7; 8)
          IDENT@[7; 8) "v"
      EQ@[8; 9) "="
      PATH_EXPR@[9; 12)
        PATH@[9; 12)
          PATH_SEGMENT@[9; 12)
            NAME_REF@[9; 12)
              IDENT@[9; 12) "Vec"
              err: `expected SEMI`
              err: `expected expression`
    EXPR_STMT@[12; 13)
      ERROR@[12; 13)
        COLON@[12; 13) ":"
        err: `expected SEMI`
        err: `expected expression`
    EXPR_STMT@[13; 14)
      ERROR@[13; 14)
        COLON@[13; 14) ":"
        err: `expected SEMI`
    EXPR_STMT@[14; 20)
      CALL_EXPR@[14; 19)
        PATH_EXPR@[14; 17)
          PATH@[14; 17)
            PATH_SEGMENT@[14; 17)
              NAME_REF@[14; 17)
                IDENT@[14; 17) "new"
        ARG_LIST@[17; 19)
          L_PAREN@[17; 18) "("
          R_PAREN@[18; 19) ")"
      SEMI@[19; 20) ";"
    EXPR_STMT@[20; 33)
      METHOD_CALL_EXPR@[20; 32)
        PATH_EXPR@[20; 21)
          PATH@[20; 21)
            PATH_SEGMENT@[20; 21)
              NAME_REF@[20; 21)
                IDENT@[20; 21) "v"
        DOT@[21; 22) "."
        NAME_REF@[22; 26)
          IDENT@[22; 26) "push"
        ARG_LIST@[26; 32)
          L_PAREN@[26; 27) "("
          LITERAL@[27; 31)
            INT_NUMBER@[27; 31) "1u32"
          R_PAREN@[31; 32) ")"
      SEMI@[32; 33) ";"
    EXPR_STMT@[33; 43)
      METHOD_CALL_EXPR@[33; 42)
        PATH_EXPR@[33; 34)
          PATH@[33; 34)
            PATH_SEGMENT@[33; 34)
              NAME_REF@[33; 34)
                IDENT@[33; 34) "v"
        DOT@[34; 35) "."
        NAME_REF@[35; 39)
          IDENT@[35; 39) "push"
        ARG_LIST@[39; 42)
          L_PAREN@[39; 40) "("
          LITERAL@[40; 41)
            INT_NUMBER@[40; 41) "2"
          R_PAREN@[41; 42) ")"
      SEMI@[42; 43) ";"
    PATH_EXPR@[43; 44)
      PATH@[43; 44)
        PATH_SEGMENT@[43; 44)
          NAME_REF@[43; 44)
            IDENT@[43; 44) "v"
    R_CURLY@[44; 45) "}"
```
which chokes on the `Vec::new()` part. @edwin0cheng @matklad

**Comment by Lapz** [comment]
I'm not sure it occurs anywhere else though

**Comment by matklad** [comment]
Looks like we split `::` into two tokens in macros, but not in from-text parsing. Yeah, we should just refactor out multisymbol tokens

**Comment by edwin0cheng** [comment]
@Lapz which commit you are using ? Is it master branch ? I should fix this bug in #1175. 

@matklad yeah, i am investigating another bugs related to punct ... which related to multisymbol tokens ... Let me finish the investigation first and let us discuss about it.

**Comment by Lapz** [comment]
@edwin0cheng Im on a older commit. I'll rebase on to the lastest master and see if the bugs fixed

**Comment by edwin0cheng** [comment]
@Lapz thanks a lot

**Comment by Lapz** [comment]
@edwin0cheng i tried it and i still get the same error

**Comment by edwin0cheng** [comment]
@Lapz Okay, would you mind to tell me what is the argument of the macro invocation ?

**Comment by matklad** [comment]
@edwin0cheng @Lapz it might be more convenient to use Zulip for such more fluent conversations :)

**Comment by Lapz** [comment]
> @Lapz Okay, would you mind to tell me what is the argument of the macro invocation ?

it was vec![1u32,2];

**Comment by kiljacken** [comment]
This was fixed in #1193, time to close?

**Comment by Lapz** [comment]
Yep
